src/sentry/eventstream/snuba.py

```diff
@@ -161,6 +161,24 @@ def insert(
         event_type = self._get_event_type(event)
         occurrence_data = self._get_occurrence_data(event)
 
+        # instead of normalizing and doing custom 'contexts' processing in snuba, we elect to do it here instead to
+        # avoid having to clutter up snuba code with business logic
+        if event_type == EventStreamEventType.Generic:
+            if event_data.get("contexts"):
+                event_data = dict(event_data)
+                contexts = event_data.setdefault("contexts", {})
```

Comment:
feel like we don't need the `setdefault` here given that we already tested for existence on line 167. maybe just move into a variable and change to `if contexts:`